src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -182,6 +188,46 @@ class CompilationState( statements : DeepDiveLog.Program, config : DeepDiveLog.C
     }
   }
 
+  // resolve an expression
+  def compileExpr(e: Expr, cq: ConjunctiveQuery, alias: AliasStyle, index : Int, isHead: Boolean) : String = {
+    def recurse(e: Expr, alias: AliasStyle, depth: Int) : String = {
+      // for columns without a name (constant, function call, binary operator), add an column index alias if necessary
+      val columnAlias = if (depth == 0 && isHead) s" AS column_${index}" else ""
```

Comment:
Unless `depth` here can be used otherwise, you can put this `" AS ..."` appending right under `compileExpr` that calls this inner recursive function.  The name `recurse` is bad and should be renamed to something like `compileExprInner` or so.

Is the alias argument also necessary?  Could you comment why it's sometimes overriden with `OriginalOnly`?